algorithms/src/Kokkos_Random.hpp

```diff
@@ -1543,13 +1543,23 @@ template <class ViewType, class RandomPool, class IndexType = int64_t>
 void fill_random(ViewType a, RandomPool g,
                  typename ViewType::const_value_type begin,
                  typename ViewType::const_value_type end) {
-  fill_random(typename ViewType::execution_space{}, a, g, begin, end);
+  Kokkos::fence(
+      "fill_random: fence before since no execution space instance provided");
+  typename ViewType::execution_space exec;
+  fill_random(exec, a, g, begin, end);
+  exec.fence(
+      "fill_random: fence after since no execution space instance provided");
 }
 
 template <class ViewType, class RandomPool, class IndexType = int64_t>
 void fill_random(ViewType a, RandomPool g,
                  typename ViewType::const_value_type range) {
-  fill_random(typename ViewType::execution_space{}, a, g, 0, range);
+  Kokkos::fence(
+      "fill_random: fence before since no execution space instance provided");
+  typename ViewType::execution_space exec;
+  fill_random(exec, a, g, 0, range);
+  exec.fence(
+      "fill_random: fence after since no execution space instance provided");
```

Comment:
Why `exec.fence()` rather than `Kokkos::fence()` like I `deep_copy` does?

Reply:
What benefit would calling `Kokkos::fence` again have? We only submit work to a specific execution space instance. Any prior work is already caught in the first `Kokkos::fence`.

Reply:
Consistency in our semantics

Reply:
Right, but we have other places where we deviate from calling `Kokkos::fence` before and after already. I would also argue that from a users-perspective there is practically no difference except for `Kokkos::fence` to have more overhead.

Reply:
Withdrawn.